### PR TITLE
Convert module to use Litmus

### PR DESCRIPTION
**Pull request by jarretlavallee**
This WIP PR to validate the viability of converting this module to use Litmus instead of beaker for the acceptance testing in Travis. ~~There are some hacky things that I needed to do to get it to work, and it is not integrated into the .sync.yaml yet. https://github.com/puppetlabs/pdk-templates/pull/268 will simplify the configuration.~~

This PR is aimed at looking into the timing and basic functionality. A few things that need to be addressed before moving forward with this conversion are below. 

- [x] Is there a better way to install the certificates on the node?
- [x] Should/can we move the bolt commands into the `spec_helper_acceptance_local.rb` to simplify the logic?
- [x] Do the sets in `provision.yaml` make sense?
- [x] How much more time does this add to the testing in Travis?

### CODE REVIEW

**Comment by jarretlavallee** [comment]
The Travis testing time went from 13 minutes to 21 minutes and 45 seconds.

Previously we were testing Puppet 6 with the following.
* Centos 7
* Ubuntu 16.04 

This change tests Puppet 5 and Puppet 6 on the following.
* Centos 7
* Centos 6
* Oracle Linux 7
* Scientific Linux 7
* Ubuntu 16.04
* Ubuntu 18.04
* Debian 8
* Debian 9

So we went from 2 tests to 16 tests and increased the total time by 8.75 minutes. The tests are over more operating systems and test 2 versions of puppet against them.

I'll remove Oracle and Scientific linux to see if it helps decrease the times.

**Comment by jarretlavallee** [comment]
After removing Scientific and oracle, the test is down to 15 min 33 sec, which is double the original numbers for 12 tests.

**Comment by geoffnichols** [comment]
@jarretlavallee I noticed this hasn't seen any activity in about a month, do you think you'll be able to work on it in the near future?

No pressure, just checking in.

**Comment by jarretlavallee** [comment]
@geoffnichols we can mark this one for the backlog. I'll probably start looking at this again next month.

**Comment by tkishel** [comment]
Charlie suggested that the puppet-metrics-viewer, which uses Docker, could be modified to be the Litmus test for the puppet_metrics_dashboard AND replace the puppet-metrics-viewer, allowing us to consolidate down to one metrics project???

(Outside my capacity at this time)

**Comment by jarretlavallee** [comment]
This should be working now. We will see how long the tests take, but it should be ~20 minutes.

**Comment by jarretlavallee** [comment]
15 minutes 43s for the acceptance tests

**Comment by suckatrash** [comment]
@jarretlavallee Looks great! Could you squash the commits and update the changelog before this goes in?
I haven't used litmus yet but the provisioning and installation commands look really useful.

**Comment by jarretlavallee** [comment]
Squashed and the changelog has been updated. If this is merged it will resolve #84.

File: spec/spec_helper_acceptance_local.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  c.before :suite do
+    # Fake the SSL certificates
+    run_shell('mkdir -p /etc/puppetlabs/puppet/ssl/{certs,private_keys}')
+    run_shell('touch $(/opt/puppetlabs/bin/puppet config print localcacert) \
+                $(/opt/puppetlabs/bin/puppet config print hostcert) \
+                $(/opt/puppetlabs/bin/puppet config print hostprivkey)')
+    pp = <<-PUPPETCODE
+      package{'toml-rb':
+        ensure   => installed,
+        provider => puppet_gem,
```
[the remark this answers is not in the document]
Comment (jarretlavallee) [reply spec/spec_helper_acceptance_local.rb:12]: Yup. Its only testing puppet apply in this scenario. We can expand the acceptance test use cases as I only ported them over in this case.

File: provision.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  images: ['centos/7', 'generic/ubuntu1804']
+travis_deb:
+  provisioner: docker
+  images: ['litmusimage/ubuntu:16.04', 'litmusimage/ubuntu:18.04', 'litmusimage/debian:8', 'litmusimage/debian:9']
+travis_el:
+  provisioner: docker
+  images: ['litmusimage/centos:6', 'litmusimage/centos:7' ]
+release_checks:
+  provisioner: vmpooler
+  images: ['redhat-5-x86_64', 'redhat-6-x86_64', 'redhat-7-x86_64', 'redhat-8-x86_64', 'centos-5-x86_64', 'centos-6-x86_64', 'centos-7-x86_64',  'centos-8-x86_64', 'oracle-6-x86_64', 'oracle-7-x86_64', 'scientific-6-x86_64', 'scientific-7-x86_64', 'debian-8-x86_64', 'debian-9-x86_64',  'debian-10-x86_64', 'ubuntu-1404-x86_64', 'ubuntu-1604-x86_64', 'ubuntu-1804-x86_64', 'sles-11-x86_64', 'sles-12-x86_64', 'sles-15-x86_64']
```
[the remark this answers is not in the document]
Comment (jarretlavallee) [reply provision.yaml:16]: Good call. I'll remove them as we will never use them. This is a list of potential options for running the testing in vmpooler and would need to be run manually on a configured workstation.

[the remark this answers is not in the document]
Comment (jarretlavallee) [reply provision.yaml]: Done in the lastest force push

[the remark this answers is not in the document]
Comment (jarretlavallee) [reply provision.yaml]: Note that I left in EL 8 and debian 10 for testing as we may want to develop on those in the future.

**Pull request merged**